### PR TITLE
Document the return value of AccessDialog()

### PR DESCRIPTION
**Pull request by WhyNotHugo**
Closes: https://github.com/flatpak/xdg-desktop-portal/issues/1041

### CODE REVIEW

File: data/org.freedesktop.impl.portal.Access.xml [hunk 1]
```diff
@@ -37,7 +37,8 @@
         @subtitle: Subtitle for the dialog
         @body: Body text, may be ""
         @options: Vardict with optional further information
-        @response: Numeric response
+        @response: Numeric response. A value of 0 indicates that the user has
+          agreed to grant permission. Any other value is treated as a refusal.
```
Comment (whot) [inline data/org.freedesktop.impl.portal.Access.xml:41]: (almost) all the other `impl.portal` just state `Numeric response` too - it'd be nice to update all those, and ideally to something like: "The value returned as the response argument in the #org.freedesktop.portal.Request::Response for the corresponding Request". This way you get the automatic link to where the documentation for the return values is.

**Review by whot — CHANGES_REQUESTED** [review]

File: data/org.freedesktop.impl.portal.Access.xml (hunk 1 above)
Comment (WhyNotHugo) [reply data/org.freedesktop.impl.portal.Access.xml:41]: I'm not sure I understand. What is _"the #org.freedesktop.portal.Request::Response for the corresponding Request"_ in this case?

Comment (whot) [reply data/org.freedesktop.impl.portal.Access.xml:41]: Huh, indeed, this is one (of the few?) that doesn't have an immediately corresponding request in the client API. I think in that case something like "The value range of the response matches the values allowed for #org.freedesktop.portal.Request::Response" for this one. Still would be nice to update the others though.

Comment (WhyNotHugo) [reply data/org.freedesktop.impl.portal.Access.xml]: Updated.

**Comment by WhyNotHugo** [comment]
It's not rendering correctly; I'm not entirely sure why.

**Comment by WhyNotHugo** [comment]
Looks like the `#org.freedesktop.portal.Request::Response` doesn't actually add a link tho.

I'm not entirely sure what format this is, so I'm not sure where to look for its documentation. How do I add an actual link to the anchor?

**Comment by WhyNotHugo** [comment]
Oh, there's an example right above it :facepalm: 

I tried this:

```
diff --git a/data/org.freedesktop.impl.portal.Access.xml b/data/org.freedesktop.impl.portal.Access.xml
index 29a9f70..5759e02 100644
--- a/data/org.freedesktop.impl.portal.Access.xml
+++ b/data/org.freedesktop.impl.portal.Access.xml
@@ -37,7 +37,7 @@
         @subtitle: Subtitle for the dialog
         @body: Body text, may be ""
         @options: Vardict with optional further information
-        @response: Numeric response. The values allowed match the values allowed for #org.freedesktop.portal.Request::Response signal.
+        @response: Numeric response. The values allowed match the values allowed for <link linkend="#gdbus-signal-org-freedesktop-portal-Request.Response">#org.freedesktop.portal.Request::Response</link> signal.
         @results: Vardict with the results of the call
 
         Presents a "deny/grant" question to the user.
```

But it fails to build:

```
ninja: job failed: /usr/bin/xmlto xhtml-nochunks -o doc -m ../doc/xmlto-config.xsl --searchpath /home/hugo/src/github.com/flatpak/xdg-desktop-portal/build/src doc/portal-docs.xml
xmlto: /home/hugo/src/github.com/flatpak/xdg-desktop-portal/build/doc/portal-docs.xml does not validate (status 3)
xmlto: Fix document syntax or use --skip-validation option
/home/hugo/src/github.com/flatpak/xdg-desktop-portal/build/doc/portal-docs.xml:133: element link: validity error : Syntax of value for attribute linkend of link is not valid
/home/hugo/src/github.com/flatpak/xdg-desktop-portal/build/doc/portal-docs.xml:133: element link: validity error : IDREF attribute linkend references an unknown ID "#gdbus-signal-org-freedesktop-portal-Request.Response"
Document /home/hugo/src/github.com/flatpak/xdg-desktop-portal/build/doc/portal-docs.xml does not validate
ninja: subcommand failed
```

I'm not sure how to work around this. Any insight is welcome.

**Comment by GeorgesStavracas** [comment]
We're reworking the docs system on #1157 but this is good to have already

**Comment by WhyNotHugo** [comment]
Sphinx sound like a great choice :+1:

**Pull request merged**